src/components/app-switcher/app-switcher.js

```diff
@@ -0,0 +1,201 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Renders an application drawers that can be used to switch programs
+ * @class AppSwitcher
+ */
+
+export default class AppSwitcher extends React.Component {
+
+  static propTypes = {
+        /**
+         * The json blob for the application links
+         *
+         * @property applicationJson
+         * @type {String}
+         * @default ''
+         */
+    applicationJson: PropTypes.string.isRequired,
+
+        /**
+         * The title of the menu item
+         *
+         * @property menuTitle
+         * @type {String}
+         * @default ''
+         */
+    menuTitle: PropTypes.string
+
+  }
+
+    /**
+     * Return the classes for the styles
+     *
+     * @method styles
+     * @return Hash
+     */
+
+  get styles() {
+    const prefix = 'react-switcher-';
+    return {
+      appItem: prefix + "app-item",
+      menu: classNames((this.state.active ? prefix + "menu-active" : ""), prefix + "menu-title"),
+      sectionTitle: prefix + "section-title",
+      drawer: prefix + "drawer",
+      arrowUp: prefix + "arrow-up",
+      symbol: prefix + "symbol"
+    };
+  }
+
+    /**
+     * Constructor method setting properties
+     *
+     * @method constructor
+     * @return {Void}
+     */
+
+  constructor(...props) {
+    super(...props);
+
+    this.state = {
```

Comment:
There's no need for state here, the data would be kept at a higher level that the component and provided by that framework (which would be flux in our apps). 

Carbon components as a rule should contain as little state as possible - the state is provided by the apps that use Carbon.